src/network/networkd-mptcp.c

```diff
@@ -0,0 +1,337 @@
+/* SPDX-License-Identifier: LGPL-2.1+
+ * Copyright © 2020 VMware, Inc. */
+
+#include <linux/mptcp.h>
+
+#include "alloc-util.h"
+#include "netlink-util.h"
+#include "networkd-manager.h"
+#include "networkd-mptcp.h"
+#include "parse-util.h"
+#include "set.h"
+#include "string-util.h"
+
+static int mp_tcp_new(MPTCP **ret) {
+        MPTCP *mp_tcp;
+
+        mp_tcp = new(MPTCP, 1);
+        if (!mp_tcp)
+                return -ENOMEM;
+
+        *mp_tcp = (MPTCP) {
+                  .family = AF_UNSPEC,
+                  .id_is_set = false,
```

Comment:
Not a matter, but these lines are redundant. `new0` is enough.

Reply:
It's good to init to known values. Just because AF_UNSPEC and false is zero it does mean we skip it.